packages/backend-app-api/src/wiring/BackendInitializer.ts

```diff
@@ -74,12 +83,18 @@ const instanceRegistry = new (class InstanceRegistry {
 
   #exitHandler = async () => {
     try {
-      const results = await Promise.allSettled(
-        Array.from(this.#instances).map(b => b.stop()),
-      );
-      const errors = results.flatMap(r =>
-        r.status === 'rejected' ? [r.reason] : [],
-      );
+      // This signals to the healthcheck service that the process is shutting down
+      process.exitCode = 0;
+
+      const results = await Promise.race([
+        // Give the backend 30 seconds to shut down, then force exit
+        new Promise(resolve => setTimeout(resolve, 30000)),
+        Promise.allSettled(Array.from(this.#instances).map(b => b.stop())),
+      ]);
```

Comment:
I don't think this is quite the place to handle this kind of force exit logic tbh, better have that live within the backend instances instead imo

Reply:
We realized that it is already handled by the plugins lifecycle middleware:
https://github.com/backstage/backstage/blob/2b6efe53674316b90f0f11a04e417bad78b64631/packages/backend-defaults/src/entrypoints/httpRouter/http/createLifecycleMiddleware.ts#L97-L101